src/librustdoc/html/render/print_item.rs

```diff
@@ -1869,7 +1869,11 @@ fn document_type_layout(w: &mut Buffer, cx: &Context<'_>, ty_def_id: DefId) {
         return;
     }
 
-    writeln!(w, "<h2 class=\"small-section-header\">Layout</h2>");
+    writeln!(
+        w,
+        "<h2 id=\"layout\" class=\"small-section-header\"> \
+        Layout<a href=\"#layout\" class=\"anchor\"></a></h2>"
```

Comment:
Also, the anchor is supposed to be before, not after.

Reply:
It's the same in other sections:

https://github.com/rust-lang/rust/blob/2f320a224e827b400be25966755a621779f797cc/src/librustdoc/html/render/print_item.rs#L692

https://github.com/rust-lang/rust/blob/2f320a224e827b400be25966755a621779f797cc/src/librustdoc/html/render/print_item.rs#L1118